### DISCUSSION TITLE
The ReadME Project is up for discussion!

### DISCUSSION THREAD

**Discussion by mikethemelanson** [community/community#55584] [Discover: GitHub Best Practices]
![The ReadME Project logo](https://user-images.githubusercontent.com/8250055/101071884-0efd3e00-3552-11eb-8523-11305d3cc848.png)
***
Hey folks! 👋 

@mikethemelanson, senior editor with GitHub’s [The ReadME Project](https://github.com/readme), here to let you know that we just published our May edition (including two featured articles by yours truly), and we wanted to give you the highlights and invite your thoughts and feedback!

This is our first post here on GitHub’s Community Discussions, but we expect many more to come. The ReadME Project is a GitHub platform dedicated to highlighting the best from the open source software community and we want to hear directly from you about the topics you want us to report on, questions you have, and folks within the software development community that you would like to learn from. If you want to find more of our posts here, just look for us using [The ReadME Project label](https://github.com/community/community/discussions?discussions_q=label%3A%22The+ReadME+Project%22).

In May’s first feature story, I explored the idea of contributor relations in open source by looking at [the story of the open source project that closed itself down entirely to code contributions](https://github.com/readme/featured/how-open-is-open-source?src=rmpcod), from features all the way to bug fixes. I not only spoke to Ben Johnson (@benbjohnson), the maintainer of that project, but also open source advocate Julia Ferraioli (@juliaferraioli), developer Shawn Wang (@sw-yx), [Prometheus](https://github.com/prometheus/prometheus) maintainer Bartek Plotka (@bwplotka), and Roberto Ierusalimschy (@roberto-ieru), one of the creators of [Lua](https://github.com/lua), to find out how open source isn’t one size fits all. 

**@benbjohnson on why he chose to close his project to code contributions:** `”Even small contributions typically required hours of my time to properly test and validate them. I am grateful for community involvement and when folks report bugs or suggest features. I do not wish to come off as anything but welcoming, however, I've made the decision to keep this project closed to contribution for my own mental health and long-term viability of the project.”`

Next, I met Paul Chiou (@oikos99), a developer pursuing his Ph.D in computer science who builds software to automatically detect and fix keyboard accessibility issues. The thing is, Paul is paralyzed from the neck down and can't use a physical keyboard - instead, he designed and built custom hardware and software to act as his mouse. When Paul wants to do something like use a computer, or perhaps play some Dota 2, [he gets creative](https://github.com/readme/featured/disability-creativity?src=rmpcod). Beyond creativity, Paul is working to make the world more accessible for everyone.

**Paul on accessibility as a developer:** `“I believe every software developer has the obligation to make the software they create accessible. Just like software testing is important, and no developer would deploy their products without testing for bugs, I believe the same should be applied to accessibility bugs.”`

### We also dive into TypeScript tooling, building software with generative AI, and using GitHub Copilot for MLOps, TDD, and more - check it out!

- Ever get formatters, linters, and compilers mixed up? [This guide by Josh Goldberg (@JoshuaKGoldberg) will set you straight](https://github.com/readme/guides/formatters-linters-compilers?src=rmpcod).
- Python's not the only game in town. Find out why Noah Gift (@noahgift) argues that  [Rust + GitHub Copilot could be a winning combination for MLOps](https://github.com/readme/guides/rust-mlops?src=rmpcod).
- Learn all about [how to build software with generative AI](https://github.com/readme/guides/coding-generative-ai). Software developer Anton Mirhorodchenko (@GreenWizard2015) explains how he uses LLMs for everything from architecture and design down to coding the nitty gritty.
- You could also [accelerate test-driven development with AI](https://github.com/readme/guides/github-copilot-automattic?src=rmpcod) by writing your tests and letting GitHub Copilot create the corresponding code, explains Gio Lodi (@mokagio).
- And we also hear from Annalu Waller, who [champions accessibility in her efforts to unleash the untapped potential in everyone](https://github.com/readme/stories/annalu-waller?src=rmpcod) 
- Last up, the [latest episode of The ReadME Podcast](https://github.com/readme/podcast/open-closed-equilibrium?src=rmpcod) also dropped, where we talk a bit more about methods maintainers might take regarding contributions, and we hear from Avid Press (@avipress) about the benefits of analytics to maintainers and the open source community.

❓We’re always looking for new topics to write about, and we want to hear from **YOU**. Where do you think we should explore in the coming months? Do you want to hear more about how to use GitHub Copilot? How about GitHub Actions? Or are there certain programming languages or paradigms you’d like to hear more about? Or perhaps it’s time to dive into the back end and get into the operations side of things? Where should The ReadME Project team explore next?
***
To get the latest stories, articles, and best-practices from The ReadME Project delivered directly to your inbox, [subscribe to The ReadME Project newsletter](https://github.com/readme/?mc_cid=acb0681fa4&mc_eid=a111345150#newsletter).

**Comment by sibaonikk** [comment]
hello

**Comment by KambonoJeff** [comment]
**Where do you think we should explore in the coming months?**

   Hey 👋👋👋👋

Here are some of the topics you could expound on...

✍️ **DevOps and Site Reliability Engineering (SRE):** Focus on the operations side of things by exploring DevOps practices, SRE methodologies, and how they contribute to building scalable and reliable software systems.

✍️ **Serverless computing**  : Shed light on the serverless paradigm, its advantages, and practical use cases. Discuss the major serverless platforms, architectural patterns, and best practices.

✍️ **Progressive web applications (PWAs):** Explore the concept of PWAs, their benefits, and how they are transforming web development. Showcase successful PWA implementations and provide guidance on building PWAs effectively.

✍️ **Microservices architecture:** Delve into the world of microservices and discuss its benefits, challenges, and best practices. Explore popular frameworks and tools used in building and deploying microservices.

✍️ **Data engineering and analytics:** Explore the realm of data engineering, data pipelines, and analytics. Discuss different data processing frameworks, data lakes, real-time analytics, and the impact of data-driven decision making.

**Reply by mikethemelanson** [reply]
Thanks for sharing, @KambonoJeff! 👍 

One place to keep an eye out for our stories in this general realm is [the DevOps topic](https://github.com/readme/topics/devops?src=rmpcod) on The ReadME Project. Many of these are realms I've been watching for the last several years, so I'll definitely keep them in mind. 

On the DevOps/SRE end of things, I wonder - did you catch my [featured article about GitOps](https://github.com/readme/featured/defining-gitops?src=rmpcod) earlier this year? 

And for the above suggested topics, do you happen to have any favorite open source projects that you like to pay attention to?

**Comment by Alaaryad** [comment]
thanx
